google-pubsub/1.0/modules/ROOT/pages/google-pubsub-connector-reference.adoc

```diff
@@ -0,0 +1,641 @@
+|===
+
+[[RedeliveryPolicy]]
+=== Redelivery Policy
+
+Configures the redelivery policy for executing requests that generate errors. You can add a redelivery policy to any source in a flow.
+
+[%header,cols="20s,25a,30a,15a,10a"]
+|===
+| Field | Type | Description | Default Value | Required
+| Max Redelivery Count a| Number | Maximum number of times that a redelivered request can be processed unsuccessfully before returning a REDELIVERY_EXHAUSTED error. |  |
+| Message Digest Algorithm a| String | Secure hashing algorithm to use if the *Use Secure Hash* field is `true`. If the payload of the message is a Java object, Mule ignores this value and returns the value that the payload’s `hashCode()` returned. |  |
+| Message Identifier a| <<RedeliveryPolicyMessageIdentifier>> | Defines which strategy is used to identify the messages. |  |
+| Object Store a| ObjectStore | Configures the object store that stores the redelivery counter for each message.  |  |
+|===
+
+[[RedeliveryPolicyMessageIdentifier]]
+=== Redelivery Policy Message Identifier
+
+Configures how to identify a redelivered message and how to find out when the message was redelivered.
+
+[%header,cols="20s,25a,30a,15a,10a"]
+|===
+| Field | Type | Description | Default Value | Required
+| Use Secure Hash a| Boolean | If `true`, Mule uses a secure hash algorithm to identify a redelivered message. |  |
+| Id Expression a| String | One or more expressions that determine when a message was redelivered. You can set this property only if the *Use Secure Hash* field is `false`. |  |
+|===
+
+[[Snapshot]]
+=== Snapshot
+
+Snapshot resource. Use snapshots in seek operations, which manage message acknowledgments in bulk.
+This type enables you to set the acknowledgment state of messages in an existing subscription to the state captured by a snapshot.
+
+
+[%header,cols="20s,25a,30a,15a,10a"]
+|===
+| Field | Type | Description | Default Value | Required
+| Name a| String | Name of the snapshot. |  |
+| Topic a| String | Name of the topic from which the snapshot retains messages. |  |
+| Expire Time a| DateTime | Time that the snapshot is guaranteed to exist up until. A newly-created snapshot expires no later than seven days from the time of its creation. Its exact lifetime is determined at creation by the existing backlog in the source subscription. The lifetime of the snapshot is seven days, which is the age of the oldest unacked message in the subscription. Consider a subscription with an oldest unacked message that is three days old. If a snapshot is created from this subscription, the snapshot, which always captures this three day old backlog as long as the snapshot exists, will expire in four days. The service will refuse to create a snapshot that will expire in less than one hour after creation. A timestamp is in RFC3339 UTC Zulu format, with nanosecond resolution and up to nine fractional digits, such as 2014-10-02T15:01:23Z and 2014-10-02T15:01:23.045123456Z. |  |
```

Comment:
Tough to understand. I think it would be clearer if you separate it into paragraphs. The way I do this for table entries is as follows:

{sp} +
{sp} +